AwsCryptographicMaterialProviders/dafny/AwsCryptographyKeyStoreAdmin/src/Mutations.dfy

```diff
@@ -149,6 +166,59 @@ module {:options "/functionSyntax:4" } Mutations {
     return Success(ActiveVerificationHolder.NotDecrypt());
   }
 
+  method {:isolate_assertions} VerifyEncryptedHierarchicalKeyForHV2(
+    nameonly item: Types.AwsCryptographyKeyStoreTypes.EncryptedHierarchicalKey,
+    nameonly keyManagerStrategy: KmsUtils.keyManagerStrat,
+    nameonly localOperation: string,
+    nameonly mutationToApply: StateStrucs.MutationToApply
+  )
+    returns (output: Result<ActiveVerificationHolder,Types.Error>)
+
+    requires mutationToApply.ValidState()
+    requires Structure.EncryptedHierarchicalKeyFromStorage?(item)
+    requires KmsArn.ValidKmsArn?(item.KmsArn)
+    requires keyManagerStrategy.ValidState()
+    requires KmsUtils.IsSupportedKeyManagerStrategy(mutationToApply.Terminal.hierarchyVersion, keyManagerStrategy)
+    requires item.Type.ActiveHierarchicalSymmetricVersion? || item.Type.HierarchicalSymmetricVersion?
+    requires mutationToApply.Terminal.hierarchyVersion.v2?
+    modifies keyManagerStrategy.Modifies
+    ensures keyManagerStrategy.ValidState()
+  {
```

Comment:
We don't need `:- Need(keyManagerStrategy.kmsSimple?, Types.UnsupportedFeatureException(message:=KeyStoreAdminErrorMessages.UNSUPPORTED_KEY_MANAGEMENT_STRATEGY_HV_2));` because its already done by upstream method. 

Post condition `requires KmsUtils.IsSupportedKeyManagerStrategy(mutationToApply.Terminal.hierarchyVersion, keyManagerStrategy)` requires it to be done